### ISSUE TITLE
Allow TaxoPress widget to support heirarchy view

### ISSUE THREAD

**Issue by stevejburge** [TaxoPress/TaxoPress#1109]
The default "Categories" widget / block has a "Show hierarchy" feature
<img width="1139" alt="Screen Shot 2021-11-29 at 12 56 01 PM" src="https://user-images.githubusercontent.com/3868028/143918838-27d01edc-f9dc-4a03-8661-6fd660460653.png">

We should make that possible also. The setting should appear if "List UL/LI" is selected

<img width="853" alt="Screen Shot 2021-11-29 at 12 57 13 PM" src="https://user-images.githubusercontent.com/3868028/143918948-f22dc37a-99f6-4d4b-8933-15733daab4e2.png">

**Comment by ojopaul** [comment]
@stevejburge this issue is refering to "Terms Display" settings output and where again, what's the name of the widget you screenshot ?

**Comment by stevejburge** [comment]
@ojopaul this our widget here:
<img width="401" alt="Screen Shot 2021-12-02 at 9 41 13 AM" src="https://user-images.githubusercontent.com/3868028/144443676-0cdcad26-86a0-4e97-8ed9-c20320c0465a.png">

**Comment by ojopaul** [comment]
@stevejburge , while working on this, i discovered "Terms Display" actually show terms/tags and not taxonomy/category that has hierarchy feature. So, how's hierarchy suppose to work since it's not listing taxonomy/category ? 

![cat](https://user-images.githubusercontent.com/92122385/144606768-c1deb2fa-42ad-4903-b906-99af9f8c0978.jpg)

**Comment by stevejburge** [comment]
@ojopaul Yes, that's the default, but you can change that setting to use Categories instead:
<img width="754" alt="Screen Shot 2021-12-03 at 12 06 46 PM" src="https://user-images.githubusercontent.com/3868028/144643433-bc822e60-6619-406e-9821-30e50c203dd6.png">

**Comment by ojopaul** [comment]
You still doesn't get what i was trying to point out. Check my attached screenshot. **Category** is showing taxonomy which has hierarchy feature and terms display is showing **terms** which doesn't have such feature.

Terms doesn't have hierarchy feature.

@stevejburge

**Comment by ojopaul** [comment]
Oh, i understood you now. I was checking tags. I think the main problem here is, most taxonomy are like post_tags that doesn't have hierarchy options. Reason why i wasn't getting it before.

Will check now testing with category 
@stevejburge

**Comment by ojopaul** [comment]
For example, media_tags from your screenshot doesn't have 'hierarchy' feature like category.

**Comment by ojopaul** [comment]
So, that means this feature will only works in taxonomy that has child parent relationship.

**Comment by stevejburge** [comment]
@ojopaul Sorry, some of this is a bit confusing

My categories:

<img width="680" alt="Screen Shot 2021-12-03 at 12 14 38 PM" src="https://user-images.githubusercontent.com/3868028/144644848-82f0f619-84b7-4600-9634-15e77ab4f479.png">

My settings:

<img width="587" alt="Screen Shot 2021-12-03 at 12 16 14 PM" src="https://user-images.githubusercontent.com/3868028/144644892-3e262312-8a8f-4060-94e6-51856c1f5f2d.png">

Output from the TaxoPress widget, with no hierarchy:


<img width="300" alt="Screen Shot 2021-12-03 at 12 16 23 PM" src="https://user-images.githubusercontent.com/3868028/144644946-e02ffc6f-9c08-4b4c-9dd1-c539003c217f.png">

**Comment by ojopaul** [comment]
Yes, i got it now. I was having issues before because this feature won't work for some taxonomy that doesn't have parent child relationship.

For example, post_tags, media_tags etc

**Comment by stevejburge** [comment]
Good point. Maybe there should be a setting?

"Show hierarchy? (if applicable)"

**Comment by ojopaul** [comment]
Yes, make sense

**Comment by ojopaul** [comment]
@stevejburge i'm not able to implement this feature due to ways terms display is implemented.

It's not same process as just retrieving terms same way the screenshot category is done. Tracing the source code, i noticed the main feature is tag cloud which pass though various configurations that make hierarchy impossible some and not limited to (1.) Timeframe limit(based on if a category is used and this filter is done while query tags), (2.) method (which could be random, alphabetical etc) among others. I tried changing the function when hierarchy is selected for terms display but i realised some features/user configuration are lost in the process.

If we think this through, this feature is not suitable for terms display unless we want to ignore some of the settings when hierarchy is selected and simply query same way as normal category is done.

**Comment by stevejburge** [comment]
OK, thanks @ojopaul 

I appreciate the detailed explanation

Let's hold this for next milestone which will have a big redesign for Terms Display

**Comment by ojopaul** [comment]
There's no more issue on this milestone, should I send the package or wait
for more issues to be added?

**Comment by stevejburge** [comment]
Sure, let's do it

Steve

**Comment by ojopaul** [comment]
@stevejburge have you decided for the way forward on this considering the fact it's in an active work milestone ?

**Comment by stevejburge** [comment]
@ojopaul I think we should do this

If we do this, should we disable the "Timeframe limit" and "Method" options?

**Comment by ojopaul** [comment]
@stevejburge , i've checked this again but won't be able to predict what'll work or won't work when 'heirarchy' is enabled.

Here's traditional wordPress term query parameters https://developer.wordpress.org/reference/classes/WP_Term_Query/__construct/ and any feature outside the accepted parameters, i'll only be able to determine if they'll work or not or if i'll be able to find way to make them work or not during development.

So, if this is really necessary, then we can make this to be exclusive to terms that support heirarchy (like category) and not those that are not supporting heirarchy like (tags) when UL/LI format is selected.

So, may not be able to give accurate replies on features that'll be lost in all the Terms Displays settings until i started working and try first.
